consensus/istanbul/core/handler.go

```diff
@@ -23,6 +23,14 @@ import (
 	"github.com/ethereum/go-ethereum/consensus/istanbul"
 )
 
+func (c *core) ParentSeals() *messageSet {
```

Comment:
Imagine that a node is the proposer one block after becoming a validator. They won't have seen any `Commit` messages for the previous block, since they were not a validator during that block. However, they *will* have seen a seal for the previous block representing 2F+1 validator signatures, and we should be sure to include those signatures in the `parentSeal` that we include in our proposal, otherwise the block will be invalid.

Essentially, we should always set the parent seal to be the union of "committed seal on previous block", "commits seen for previous block". Note that these can be different even in the case where we've been a validator for a while, since if someone else sees 2F+1 Commits before we do, they will broadcast the block with that committed seal, we will accept it, and move on to a new sequence.

Reply:
I agree with this approach:
> we should always set the parent seal to be the union of "committed seal on previous block", "commits seen for previous block"

The problem is the aggregation. "committed seal on previous block" is a BLS aggregated signature. The commits seen for previous block are signatures which we have not aggregated yet. In order to get the union of the seals in this block and the previous block's we have to get the seals themselves (and not their aggregate version), which I believe the node does not store currently.

Reply:
> The problem is the aggregation. "committed seal on previous block" is a BLS aggregated signature. The commits seen for previous block are signatures which we have not aggregated yet. In order to get the union of the seals in this block and the previous block's we have to get the seals themselves (and not their aggregate version), which I believe the node does not store currently (since they are stored in [`c.current.Commits`](https://github.com/celo-org/celo-blockchain/blob/0c903dd7f906bf5f4d649c4737e9b517012c8d55/consensus/istanbul/core/core.go#L178) which is not persisted)

I had a quick chat with @mstraka100 about this, and he says that it is possible to "incrementally" aggregate an aggregated BLS signature.  The operation to do so should be the same as the one that was used to generate the original aggregation.